### PR TITLE
[PI-6] Create endpoints to food

### PR DESCRIPTION
**Pull request by matikaa**

### CODE REVIEW

File: src/main/java/com/restaurant/app/food/repository/FoodJpaRepository.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.restaurant.app.food.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.List;
+
+public interface FoodJpaRepository extends JpaRepository<FoodEntity, Long> {
```
Comment (matikaa) [inline src/main/java/com/restaurant/app/food/repository/FoodJpaRepository.java:8]: empty

File: src/main/java/com/restaurant/app/food/repository/JpaWrappedFoodRepository.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        return foodJpaRepository.existsById(foodId);
+    }
+
+    @Override
+    public void deleteById(Long foodId){
+        foodJpaRepository.deleteById(foodId);
+    }
+
+    @Override
+    public Optional<FoodModel> update(FoodModel foodModel){
```
Comment (matikaa) [inline src/main/java/com/restaurant/app/food/repository/JpaWrappedFoodRepository.java]: space {

File: src/main/java/com/restaurant/app/food/service/BaseFoodService.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                .stream().anyMatch(food -> food.positionId().equals(positionId));
+    }
+
+    @Override
+    public List<Food> getAll() {
+        return foodServiceMapper.foodModelsToFoods(foodRepository.findAll());
+    }
+
+    @Override
+    public List<Food> getFoodByCategoryId(Long categoryId){
```
Comment (matikaa) [inline src/main/java/com/restaurant/app/food/service/BaseFoodService.java]: empty before { and also below

File: src/main/java/com/restaurant/app/category/service/CategoryService.java [hunk 4]
```diff
@@ -19,7 +19,7 @@ public interface CategoryService {
 
     Optional<Category> update(UpdateCategory updateCategory);
 
-    boolean existsById(Long categoryId);
+    boolean existsByCategoryId(Long categoryId);
 
     boolean existsByPositionId(Long positionId);
 }
```
Comment (MarcinFrejnagel) [inline src/main/java/com/restaurant/app/category/service/CategoryService.java:25]: empty line after }

File: src/main/java/com/restaurant/app/category/service/CategoryServiceMapper.java [hunk 5]
```diff
@@ -12,7 +12,7 @@ public interface CategoryServiceMapper {
 
     CategoryServiceMapper INSTANCE = Mappers.getMapper(CategoryServiceMapper.class);
 
-    List<Category> categoryModelsToCategories(List<CategoryModel> categoryEntity);
+    List<Category> categoryModelsToCategories(List<CategoryModel> categoryModels);
 
     Category categoryModelToCategory(CategoryModel categoryModel);
 }
```
Comment (MarcinFrejnagel) [inline src/main/java/com/restaurant/app/category/service/CategoryServiceMapper.java:18]: empty line after }

File: src/main/java/com/restaurant/app/food/controller/FoodControllerMapper.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    FoodControllerMapper INSTANCE = Mappers.getMapper(FoodControllerMapper.class);
+
+    FoodResponse foodToFoodResponse(Food food);
+
+    FoodRequestResponse foodToFoodRequestResponse(Food food);
+
+    List<FoodResponse> foodsToFoodResponses(List<Food> foods);
+
+    Food foodRequestToFood(Long categoryId, Long foodId, FoodRequest foodRequest);
+}
```
Comment (MarcinFrejnagel) [inline src/main/java/com/restaurant/app/food/controller/FoodControllerMapper.java:24]: empty line after }

File: src/main/java/com/restaurant/app/food/controller/dto/FoodListResponse.java [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package com.restaurant.app.food.controller.dto;
+
+import java.util.List;
+
+public record FoodListResponse(
+        List<FoodResponse> foodResponses
+) {
+}
```
Comment (MarcinFrejnagel) [inline src/main/java/com/restaurant/app/food/controller/dto/FoodListResponse.java:8]: empty line after }

**Review by MarcinFrejnagel — CHANGES_REQUESTED** [review]

**Review by MarcinFrejnagel — APPROVED** [review]

**Pull request merged**